purchase_order_line_image_column/models/purchase_order_line.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2021 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    image = fields.Binary(
+        string="Image",
+        attachment=True,
+        compute="_compute_image",
+        help="This field refs purchase request line's image. This field not editable.",
+    )
+
+    @api.depends("product_id")
+    def _compute_image(self):
+        for line in self:
+            line.image = False
+            for request_line in line.purchase_request_lines:
+                if request_line.image:
+                    line.image = request_line.image
```

Comment:
Does 'line.image' contain a single image? (Are you able to ignore the case that plural images in request_line?)
Then, I suggest below. comprehensions 内包表記) is generally works faster in python. and my suggestion can avoid if-sentence in double for-loop.
```python
    for line in self:
        temp = list(request_line.image for request_line in line.purchase_request_lines if request_line.image)
        line.image = temp[0] if temp else False
```

Reply:
`temp = list(request_line.image for request_line in line.purchase_request_lines if request_line.image)` may be a little overdone.
Important things in my suggestion are to define line.image after make the 'temp' variable.

Reply:
Thank you.

I edited line below. I want to make it a single for-loop method, but hard to me.

```
        for line in self:
            for request_line in line.purchase_request_lines:
                line.image = request_line.image if request_line.image else False
```

Reply:
@TomohiroAbe-Quartile
Actually a single for-loop method is hard, I agree since comprehensions contains for-loop. 
Here, core of my suggestion("if" in double for-loop should be avoided) was to remove if-sentence from the second loop.

Please define a collection variable (like 'temp') in the first loop, first of all.
Pick up images in each "request_line" and preserve these in your collection variable
Then you can get a picture you need (or False value when no pictures are there) with avoiding "if" in double for-loop.

If you were intended to preserve every pictures, line.image would be replaced by some variable like list with Many2many or One2many field.